### PR TITLE
Release 1.5.2

### PR DESCRIPTION
**Pull request by Rob9315**
fix npm publish

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,4 +22,4 @@ jobs:
       with:
         node-version: ${{ matrix.node-version }}
     - run: npm install
-    - run: npm test
```
Comment (rom1504) [inline .github/workflows/ci.yml:25]: Why did you remove npm test ?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
+name: npm-publish
+on:
+  push:
+    branches:
+      - master # Change this to your default branch
+jobs:
+  npm-publish:
+    name: npm-publish
+    runs-on: ubuntu-latest
+    steps:
```
Comment (rom1504) [inline .github/workflows/publish.yml:10]: Why did you go back to the previous version ?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Rob9315) [reply .github/workflows/ci.yml:25]: changed it out for npm publish --dry-run, which also includes test

File: package.json [hunk 3]
```diff
@@ -1,41 +1,46 @@
...
-    "typescript": "^4.1.3"
-  }
-}
+{
+  "name": "yggdrasil",
+  "version": "1.5.2",
+  "author": "Zeke Sonxx <zeke@zekesonxx.com>",
+  "description": "Mojang authentication (Yggdrasil) client",
+  "files": [
+    "lib/**/*"
```
Comment (rom1504) [inline package.json:7]: Let's not use that, just un ignore these files by creating a .npmignore file that doesn't contain lib

Comment (Rob9315) [reply package.json:7]: will do, could you explain why?

File: .github/workflows/publish.yml (hunk 2 above)
Comment (Rob9315) [reply .github/workflows/publish.yml:10]: reversed, accident

**Review by Rob9315 — COMMENTED** [review]
please re-think the .npmignore file idea

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rom1504) [reply .github/workflows/ci.yml:25]: It's broken if npm publish runs the tests. How come ?

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -1,34 +1,35 @@
-name: npm-publish
-on:
-  push:
-    branches:
-      - master # Change this to your default branch
-jobs:
-  npm-publish:
```
Comment (rom1504) [inline .github/workflows/publish.yml:7]: Can you avoid changing all the end lines ? It makes things hard to review

File: package.json (hunk 3 above)
Comment (rom1504) [reply package.json:7]: It's better to publish all the files (unless they are very big) so users can see what they use in node_modules folder

File: .github/workflows/publish.yml (hunk 4 above)
Comment (Rob9315) [reply .github/workflows/publish.yml:7]: i am sorry about the end lines, which ones do we want to use?, will stick to those from now on

Comment (Rob9315) [reply .github/workflows/publish.yml:7]: no more from now on

File: package.json [hunk 5]
```diff
@@ -1,17 +1,18 @@
 {
   "name": "yggdrasil",
-  "version": "1.5.1",
+  "version": "1.5.2",
   "author": "Zeke Sonxx <zeke@zekesonxx.com>",
   "description": "Mojang authentication (Yggdrasil) client",
   "scripts": {
+    "prepublishOnly": "npm run test",
```
Comment (rom1504) [inline package.json:7]: Why ?
Let's not make publish depends on tests. They are run already

**Review by rom1504 — APPROVED** [review]

**Comment by rom1504** [comment]
Can you do a small pr to put build in the prepare script ?
